Backend/app/dashboard/Lib/src/Domain/Action/Dashboard/Registration.hs

```diff
@@ -234,3 +256,56 @@ buildRegistrationToken personId merchantId = do
         merchantId = merchantId,
         createdAt = now
       }
+
+registerFleetOwner ::
+  ( EsqDBFlow m r,
+    EncFlow m r,
+    HasFlowEnv m r '["dataServers" ::: [Client.DataServer]]
+  ) =>
+  FleetRegisterReq ->
+  m APISuccess
+registerFleetOwner req = do
+  runRequestValidation validateFleetOwner req
+  unlessM (isNothing <$> QP.findByMobileNumber req.mobileNumber req.mobileCountryCode) $ throwError (InvalidRequest "Phone already registered")
+  fleetOwnerRole <- QRole.findByName "FLEET_OWNER" >>= fromMaybeM (RoleDoesNotExist "FLEET_OWNER") --- we have to create a role for the fleet owner
+  fleetOwner <- buildFleetOwner req fleetOwnerRole.id
+  decFleetOwner <- decrypt fleetOwner
+  merchant <-
+    QMerchant.findByShortId req.merchantId
+      >>= fromMaybeM (MerchantDoesNotExist req.merchantId.getShortId)
+  availableServers <- asks (.dataServers)
+  unless (merchant.serverName `elem` (availableServers <&> (.name))) $
+    throwError $ InvalidRequest "Server for this merchant is not available"
+  merchantAccess <- DP.buildMerchantAccess decFleetOwner.id merchant.id
```

Comment:
no need to decrypt:
` merchantAccess <- DP.buildMerchantAccess fleetOwner.id merchant.id`